conans/model/build_info.py

```diff
@@ -25,6 +25,7 @@ def __init__(self, factory):
     def __getitem__(self, key):
         if key not in self.keys():
             super(DefaultOrderedDict, self).__setitem__(key, self.factory())
+            super(DefaultOrderedDict, self).__getitem__(key).name = key
```

Comment:
This is done in order to mimic the same behavior as ``cpp_info.name`` and default names in components